protocols/migration/migration_reports_4_0_0_to_reports_3_0_0.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from protocols import reports_4_0_0 as reports_4_0_0
+from protocols import reports_3_0_0 as reports_3_0_0
+from protocols.migration.base_migration import BaseMigration
+from protocols.migration.base_migration import MigrationError
+
+
+class MigrateReports400To300(BaseMigration):
+
+    old_model = reports_4_0_0
+    new_model = reports_3_0_0
+
+    def migrate_clinical_report_rd(self, old_instance):
+        """
+        :type old_instance: reports_4_0_0.ClinicalReportRD
+        :rtype: reports_3_0_0.ClinicalReportRD
+        """
+        new_instance = self.convert_class(self.new_model.ClinicalReportRD, old_instance)
+
+        # interpretationRequestId changed to interpretationRequestID
+        new_instance.interpretationRequestID = old_instance.interpretationRequestId
+
+        # interpretationRequestAnalysisVersion can be null in version 4
+        if hasattr(old_instance, 'interpretationRequestAnalysisVersion') and old_instance.interpretationRequestAnalysisVersion is not None:
+            new_instance.interpretationRequestAnalysisVersion = old_instance.interpretationRequestAnalysisVersion
+        else:
+            new_instance.interpretationRequestAnalysisVersion = old_instance.interpretationRequestVersion
+
+        if hasattr(old_instance, 'candidateVariants') and old_instance.candidateVariants is not None:
+            new_instance.candidateVariants = self.migrate_reported_variants(old_variants=old_instance.candidateVariants)
+        if hasattr(old_instance, 'candidateStructuralVariants') and old_instance.candidateStructuralVariants is not None:
```

Comment:
why not `if old_instance.candidateStructuralVariants is not None:` ?

Reply:
because there's no validation on whether the object has the `candidateStructuralVariants` attribute, it came up in testing